### ISSUE TITLE
[CI] integTestMinioRunner fails (poorly) if docker compose is not available

### ISSUE THREAD

**Issue by tvernum** [elastic/elasticsearch#42829]
Build: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+multijob-darwin-compatibility/432

The `postProcessFixture` [task is disabled](https://github.com/elastic/elasticsearch/blob/797052631fe6177e4c898747daa2555acf9a1f87/buildSrc/src/main/java/org/elasticsearch/gradle/testfixtures/TestFixturesPlugin.java#L70) if `docker-compose` [isn't available](https://github.com/elastic/elasticsearch/blob/797052631fe6177e4c898747daa2555acf9a1f87/buildSrc/src/main/java/org/elasticsearch/gradle/testfixtures/TestFixturesPlugin.java#L182-L184), but the `minioAddress` def in the build [relies on that task having run](https://github.com/elastic/elasticsearch/blob/797052631fe6177e4c898747daa2555acf9a1f87/plugins/repository-s3/build.gradle#L165).

So if `docker-compose` isn't available (in one of the expected locations) then the build fails with an obscure error
```
01:10:20 Execution failed for task ':plugins:repository-s3:integTestMinioRunner'.
01:10:20 > Cannot get property 'test.fixtures.minio-fixture.tcp.9000' on extra properties extension as it does not exist
```

**Comment by original-brownbear** [comment]
@mark-vieira I think this relates to https://github.com/elastic/elasticsearch/issues/42680 can you take a look here?

**Comment by mark-vieira** [comment]
I believe #42745 should address this.

**Comment by jakelandis** [comment]
This has failed 131 times in the past 4 days
<img width="1331" alt="image" src="https://user-images.githubusercontent.com/976291/59122415-533d5d00-8920-11e9-8e3a-edad66e3a0c3.png">

I'm not sure https://github.com/elastic/elasticsearch/pull/42719 addressed the issue. 

cc @mark-vieira @atorok

**Comment by henningandersen** [comment]
Fails on both 7.x and master. Can reproduce locally using:
```
./gradlew :plugins:repository-s3:integTestMinioRunner
```
Failed 80+ times since Friday.

**Comment by mark-vieira** [comment]
Looks like #42745 cleared this up. We haven't had an instance of this particular issue since June 10th. The above linked CI job is still failing but for another issue which will be sorted out by #43232.

**Comment by dakrone** [comment]
There's been another couple of occurrences of this today:

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.x+multijob-windows-compatibility/os=windows-2012-r2/165/console - https://gradle-enterprise.elastic.co/s/sc5pznpn4uzvk 
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.x+multijob-windows-compatibility/os=windows-2019/165/consoleFull - https://gradle-enterprise.elastic.co/s/ddrgb2k3gqvlq

**Comment by mark-vieira** [comment]
@atorok does https://github.com/elastic/elasticsearch/pull/42745 need to be backported to `7.x`?

**Comment by mark-vieira** [comment]
@atorok are you saying that `:x-pack:snapshot-tool:thirdPartyTest` should be disabled on Windows? If so, why is that task running? It's not a dependency issue, it's that there's some logic that should disabling that task if docker isn't available, no?

**Comment by mark-vieira** [comment]
Same on non-Windows builds for platforms where Docker is unavailable.

https://gradle-enterprise.elastic.co/s/fyzvajao2564k

It seems this task should be disabled/skipped but isn't. That's why I mentioned the above PR as I thought that fixed this problem but perhaps just didn't get merged into 7.x.

**Comment by atorok** [comment]
The test fixtures plugin should be disabling everything that requires docker. 
From what I see everything that needs to be was back-ported is done so, the plugin is identical in `7.x` and `master`.

**Comment by droberts195** [comment]
This is still failing in the same way on all versions of Windows:

* 2012r2: https://gradle-enterprise.elastic.co/s/yeeiscavgi5a2
* 2016: https://gradle-enterprise.elastic.co/s/iq2jnljycop4o
* 2019: https://gradle-enterprise.elastic.co/s/qdlnesx6eanra

**Comment by mark-vieira** [comment]
This looks to be fallout from #46780 being backported to `7.x`. In that PR we introduced the `useFixture()` DSL, which we then refactored existing projects in `master` to use. However, in `master` the `:x-pack:snapshot-tool` project no longer uses the test fixtures plugin so when this got backported stuff started to blow up.

I've submitted https://github.com/elastic/elasticsearch/pull/48166 to address this.
